### PR TITLE
feat(operation): structural matrix + vector property predicates (#244)

### PR DESCRIPTION
**Pull request by Ravenwater**
Batch 1 of the properties module (#244) — the cheapest, highest-use tier:
runtime predicate queries that scan elements and need **no factorization**
(O(n) dense / O(nnz) sparse).

## New API — `namespace mtl`, free functions

**Matrix** (`operation/matrix_properties.hpp`)
- `is_square`, `is_empty`
- `is_symmetric`, `is_hermitian`
- `is_upper_triangular`, `is_lower_triangular`, `is_triangular`
- `is_diagonal`, `is_banded(A, kl, ku)`
- `is_diagonally_dominant(A[, strict])`

**Vector** (`operation/vector_properties.hpp`)
- `is_zero`
- `is_finite`, `has_nan`, `has_inf`
- `is_normalized` / `is_unit`, `is_orthogonal_to`

## Tolerance policy

- **Structural checks** (`is_symmetric`, `is_diagonal`, `is_triangular`,
  `is_banded`, `is_zero`, …) take an **absolute** `tol` defaulting to **0**
  (exact-as-constructed). Pass `tol > 0` for an approximate check on a computed
  matrix.
- **Norm-based vector checks** (`is_normalized`, `is_orthogonal_to`) use a
  **relative** tolerance defaulting to **128·eps** — a computed norm / inner
  product is rarely bit-exact. `is_orthogonal_to` uses `|<u,v>| <= tol·‖u‖·‖v‖`,
  so it is scale-free.

## Genericity

All predicates are generic over the `Matrix`/`Vector` concepts, work with
real/complex/custom element types (via `magnitude_t` + the `conj` functor), and
run on dense or sparse `(i,j)` storage. Wired into the `mtl.hpp` umbrella.

## Tests

`tests/unit/operation/test_{matrix,vector}_properties.cpp` — exact/tolerance/
non-square/empty (0×0)/1×1 edge cases; complex hermitian-vs-symmetric;
NaN/Inf; scale-invariant orthogonality. Both green locally; full suite builds
clean.

## Scope note

This is phase 1 of 4 in #244. Follow-ups: factorization-backed booleans
(`is_spd`, `is_singular`, `determinant`), spectral/condition/rank
(`condition_number`, `numerical_rank`, `spectral_radius`), and tensor predicates.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **New Features**
  - Added matrix property predicate functions covering shape checks (square/empty), symmetry and Hermitian structure, triangular/diagonal classification, bandedness, and diagonal dominance.
  - Added vector property predicate functions for zero, finiteness, detecting NaN/Inf, unit/normalized checks, and orthogonality.
  - Exposed the new operations via the primary umbrella include.

- **Tests**
  - Added unit tests for matrix and vector predicates, including tolerance behavior, complex conjugation handling, and ensuring NaN values cause structural checks to fail.

### CODE REVIEW

**Comment by Ravenwater** [comment]
@coderabbitai review

**Comment by Ravenwater** [comment]
@coderabbitai review

File: include/mtl/operation/matrix_properties.hpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+/// A == A^T (within tol). Non-square matrices are never symmetric.
+template <Matrix M>
+bool is_symmetric(const M& A, magnitude_t<typename M::value_type> tol = 0) {
+    using std::abs;
+    using size_type = typename M::size_type;
+    if (A.num_rows() != A.num_cols()) return false;
+    const size_type n = A.num_rows();
+    for (size_type i = 0; i < n; ++i)
+        for (size_type j = i + 1; j < n; ++j)
+            if (abs(A(i, j) - A(j, i)) > tol) return false;
```
[the remark this answers is not in the document]
Comment (Ravenwater) [reply include/mtl/operation/matrix_properties.hpp]: Fixed in d0363c1. All structural deviation checks now use `!(abs(dev) <= tol)` instead of `abs(dev) > tol`, so a NaN (unordered under every comparison) fails the predicate rather than being silently accepted. Applied across is_symmetric/is_hermitian/is_upper/lower_triangular/is_diagonal/is_banded and is_zero, with NaN regression tests added to both test files. (is_diagonally_dominant, is_normalized, is_orthogonal_to were already NaN-safe via their negated forms.)

File: include/mtl/operation/vector_properties.hpp [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+}
+template <typename T>
+bool scalar_is_inf(const T& x) {
+    if constexpr (std::is_floating_point_v<T>) {
+        using std::isinf; return isinf(x);
+    } else if constexpr (requires { x.real(); x.imag(); }) {
+        using std::isinf; return isinf(x.real()) || isinf(x.imag());
+    } else {
+        return false;
+    }
```
[the remark this answers is not in the document]
Comment (Ravenwater) [reply include/mtl/operation/vector_properties.hpp:53]: Fixed in d0363c1. All structural deviation checks now use `!(abs(dev) <= tol)` instead of `abs(dev) > tol`, so a NaN (unordered under every comparison) fails the predicate rather than being silently accepted. Applied across is_symmetric/is_hermitian/is_upper/lower_triangular/is_diagonal/is_banded and is_zero, with NaN regression tests added to both test files. (is_diagonally_dominant, is_normalized, is_orthogonal_to were already NaN-safe via their negated forms.)
